### PR TITLE
Add option to pass in custom test suite name for Junit formatter

### LINKED ISSUE #2265: Allow customization option for junit formatter
### 🤔 What's the problem you're trying to solve?

currently all reports generated with cucumberjs junit formatter bear the name 'cucumber-js'. It would be great to be able to customize it via format options and cli param

### ✨ What's your proposed solution?
format option with test suite name. 

### ⛏ Have you considered any alternatives or workarounds?
open for suggestions...

### 📚 Any additional context?
On a side note, it is not really clear how to configure options in a cucumber.mjs file when there are multiple formtters defined. 
```
format: [
  'json:e2e/reports/cucumber-report.json',
  'junit:e2e/reports/cucumber-report.xml',
  'summary'
];

Adding options to the json formatter and to the junit in a single formatOptions: {} entry is cumbersome. Probably best to make each formatter an object. 

format: [
  {type: json, output: 'json:e2e/reports/cucumber-report.json', options: {x: 1, y: false}},
  { type: 'junit', output:  'e2e/reports/cucumber-report.xml', options: {x: 1, z: true}},
  'summary'
];

### PR DESCRIPTION
**Pull request by astocker-scottlogic**
### 🤔 What's changed?
Added functionality which allows users to pass in custom test suite names for the Junit formatter

### ⚡️ What's your motivation? 
Fixes #2265 

### 🏷️ What kind of change is this?

- :zap: New feature (non-breaking change which adds new behaviour)

### ♻️ Anything particular you want feedback on?

I haven't added anything to the docs because I wasn't sure if there was a suitable place, but I am happy to write some documentation if it is needed :-). 
I added two unit tests, I tried to stick to the current format being used but I'm not entirely sure it's correct.
Finally there is a test failing under the 16x windows build. However it seems this test has failed in other PRs so I'm not sure it's related to the changes I've added. 

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behavior of the code
  - [x] I have added/updated tests to cover my changes.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.
- [x] Users should know about my change
  - [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Review by davidjgoss — APPROVED** [review]
Thanks for doing this! I'll merge once it's all green.

**Comment by aslakhellesoy** [comment]
Hi @astocker-scottlogic,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**
